python3 support

kirklg opened the pull request:
PR to start working through python3 support.

kirklg committed:
Add python3.4 and 3.5 to tox setup

kirklg committed:
New compat module to provide built-ins deprecated in py3

kirklg committed:
Remove long/L suffix

kirklg committed:
Import compat functions and remove python3 check

kirklg committed:
Import compat functions and replace xrange with range

kirklg committed:
Remove long compat function and replace all long types with int

kirklg committed:
Add unicode compat function

kirklg committed:
Replace generator method .next() with built-in next()

kirklg committed:
Add code to ensure PriorityQueue gets imported on either python2 or 3

kirklg committed:
Use items() over iteritems(), py2/py3 compatible

kirklg committed:
Use items() over iteritems(), py2/py3 compatible

kirklg committed:
Revert request.offset back to 2/3 compatible long

kirklg committed:
Add 2/3 compatbile long type in minicluster

kirklg committed:
Make logger import absolute, and remove L/long prefix in ERROR_BYTES

kirklg committed:
Replace f.func_name with f.__name__ as key string

kirklg committed:
Make namenode import absolut, and improve import of urlparse for 2/3 compatibility

kirklg committed:
Improve import of urlparse for 2/3 compatbility

kirklg committed:
Replace generator .next() with next built-in

kirklg committed:
Replace .has_key with "in" <dict>

kirklg committed:
Make logger import absolute in rpc_sasl

kirklg committed:
Import __future__ for python3 support

kirklg committed:
Make print statements in examples be py3 print functions

kirklg committed:
Convert py2 print statement to py3 print function

kirklg committed:
Make imports absolute and add 2/3 compatible range function

kirklg committed:
Make imports absolute and add 2/3 compatible long function

kirklg committed:
Make imports absolute and add 2/3 compatible range function in tail_test

kirklg committed:
Make imports absolute and add 2/3 compatible <generator>-next arguments to assertRaises

kirklg committed:
Make imports absolute, add 2/3 compatible <generator>-next arguments to assertRaises, and octal syntax

kirklg committed:
Make imports absolute and add 2/3 compatible <generator>-next arguments to assertRaises in chown_test

kirklg committed:
Make imports absolute and replace deprecated assertEquals with assertEqual

kirklg committed:
Replace deprecated assertEquals with assertEqual in config_test

kirklg committed:
Make imports absolute in copytolocal_test

kirklg committed:
Absolute import and 2/3 compatible <generator>-next arguments in assertRaises, count_test

kirklg committed:
Absolute import and 2/3 compatible <generator>-next arguments in assertRaises, delete_test

kirklg committed:
Absolute import and 2/3 compatible <generator>-next arguments in assertRaises, du_test

kirklg committed:
Absolute imports and 2/3 compatible exception handling

kirklg committed:
Absolute imports and replace generator method .next() with built-in next()

kirklg committed:
Absolute imports and 2/3 compatible <generator>-next arguments in assertRaises, list_test

kirklg committed:
Absolute imports and 2/3 compatible <generator>-next arguments in assertRaises, rename_test

kirklg committed:
Absolute imports and replace deprecated assertEquals with assertEqual in stat_test

kirklg committed:
Absolute import in test_test

kirklg committed:
Absolute import in touchz_test

kirklg committed:
Import __future__ for python3 support in all tests

kirklg committed:
Convert py2 print statement to py3 print function in all tests

kirklg committed:
Insert protobuf path in sys.path, fix protobuf absolute imports

kirklg committed:
Swap protobuf2 with protobuf3 in requirements.txt

kirklg commented:
Hi @wouterdebie,

Do you want to merge this particular commit? Or the the entire branch? I'm hoping to getting all my unstaged commits into this PR this week, but I'm fine with anything.

Thanks!

wouterdebie commented:
I'm interested in getting py3 support, but no rush. If you're prepping a PR, please continue to do so! 
Thanks for the help!
On May 22, 2016 22:48, Kirk Leon Guerrero notifications@github.com wrote:Hi @wouterdebie,

Do you want to merge this particular commit? Or the the entire branch? I'm hoping to getting all my unstaged commits into this PR this week, but I'm fine with anything.

Thanks!

—You are receiving this because you were mentioned.Reply to this email directly or view it on GitHub

anneschuth commented:
We're also quite interested. Our Luigi setup in Python 3 now spends _a lot_ of time booting Hadoop just to find out HDFS targets were completed.

kirklg commented:
Update: No updates as of yet, its been a pretty busy week and pycon didn't help :).

anneschuth commented:
:) thanks for letting us know!

anneschuth commented:
Hi @kirklg, don't want to rush you (really!) but I was curious whether you happened to have made any progress on this. We'll need this soon and was wondering whether you would have any (work in progress) commits I could take as a starting point ?

kirklg commented:
Hi @anneschuth, ah! thank you so much for reminding me. Today is the day I will push some commits, standby. And thank you for the patience.

kirklg committed:
make client_id and RPC_HEADER bytes

kirklg committed:
decode protobuf node.path and make bytes

make `start_after` in `_get_dir_listing` bytes to store into
reqeust.startAfter (protobuf type)

kirklg committed:
Encode return data from snakebite.client to match minicluster data

kirklg committed:
Rename bytes to _bytes in `_read_bytes`, more bytes conversions

kirklg committed:
make pre-data `bytes` and change open mode to "wb" in `_handle_copyToLocal`

kirklg committed:
py3 style exceptions and wrap `filter` in `list` to work in py3

kirklg committed:
make stdout.write py2/py3 compatible

kirklg committed:
make pre-data `bytes` in cat_test

kirklg committed:
add py2/py3 compatible `next` calls in df_test

kirklg committed:
Fix tab spacing: pep8

kirklg committed:
absolute imports in tests

kirklg commented:
pushed a bunch of commits, note that the tests will fail, and I'm still working through this.

kirklg committed:
fix typo in assertion

kirklg committed:
Write bytes new line

kirklg committed:
replace py2 with py3 tox_env in travis.yml

kirklg committed:
fix python version in .travis.yml

kirklg committed:
.message does not exist in py3, replace with .args for py2/py3

kirklg committed:
decode bytes from client to match minicluster `text', which produces a str

kirklg committed:
Fix broken tests in client_test

socket.error.errno is now read-only in py3
add SocketError class to assign appropriate errno values to `e`
attribute.

kirklg committed:
Assign stdout alias for py2/py3

sys.stdout.buffer exists only in python3 (emits bytes)
Use sys.stdout for py2 and sys.stdout.buffer for python3

gglanzani commented:
Hi @kirklg, I'd like to help you out on this, is there something in particular that needs to be done, or is test fixing _the_ thing that needs to happen?

PS: Hi @anneschuth!

gglanzani commented:
Hi @kirklg, I think the biggest issue if you get this through, will be `python-krbV`, which is not Python 3 compatible. Are there any tests covering Kerberos support?

kirklg commented:
Hi @gglanzani, thank you for your comments above, please give me a few days to re-grasp :) the code I've pushed here as I've been off the grid doing other things and I _really_ need to finish this.

As far as the `python-krbV` thing, I am not sure why I haven't bumped into this issue yet, probably because there is no coverage for kerberos currently in `snakebite`; which should be fixed ey? ;). I will go and do some research on this, also will look into `python-gssapi` as that is python 3 compatible.

Thanks again,

Poogles commented:
Hi @kirklg, not wanting to push you here but have you got any further with this?  We're quite keen on this and we don't want to re-invent the wheel.

If you've not got any further I may take what's been done here and start porting the rest?
